src/SignalR/server/Core/src/Internal/HubCallerClients.cs

```diff
@@ -8,12 +8,43 @@ internal class HubCallerClients : IHubCallerClients
     private readonly string _connectionId;
     private readonly IHubClients _hubClients;
     private readonly string[] _currentConnectionId;
+    private readonly bool _parallelEnabled;
 
-    public HubCallerClients(IHubClients hubClients, string connectionId)
+    public HubCallerClients(IHubClients hubClients, string connectionId, bool parallelEnabled)
     {
         _connectionId = connectionId;
         _hubClients = hubClients;
         _currentConnectionId = new[] { _connectionId };
+        _parallelEnabled = parallelEnabled;
+    }
+
+    private class NotParallelSingleClientProxy : ISingleClientProxy
+    {
+        private readonly ISingleClientProxy _proxy;
+
+        public NotParallelSingleClientProxy(ISingleClientProxy hubClients)
+        {
+            _proxy = hubClients;
+        }
+
+        public Task<T> InvokeCoreAsync<T>(string method, object?[] args, CancellationToken cancellationToken = default)
+        {
+            throw new InvalidOperationException("Client results inside a Hub method requires HubOptions.MaximumParallelInvocationsPerClient to be greater than 1.");
```

Comment:
Can we abort the connection and log an error if we're awaiting `MaximumParallelInvocationsPerClient` client invocations at once even though it's greater than 1? This is a huge footgun. I can see a lot of connections locking up for no apparent reason.

Reply:
Planning on having a list of known items on the original issue once this is merged.